pkg/kubelet/kubelet.go

```diff
@@ -3282,7 +3311,21 @@ func GetPhase(spec *api.PodSpec, info []api.ContainerStatus) api.PodPhase {
 
 func (kl *Kubelet) generateAPIPodStatus(pod *api.Pod, podStatus *kubecontainer.PodStatus) api.PodStatus {
 	glog.V(3).Infof("Generating status for %q", format.Pod(pod))
+
+	// any pod that requires eviction must be added to the list to sync.
+	attrs := &lifecycle.PodEvictAttributes{Pod: pod}
+	for _, podEvictor := range kl.podEvictors {
+		if result := podEvictor.Evict(attrs); result.Evict {
```

Comment:
> I think I want to just update this to have syncPod check for the failed status and immediately kill the pod, and then call it a day and we iterate as we integrate the other evictors.

That is reasonable to me.  I think it will be tough to divine out the exact right factoring through conversation and iterating will be more productive.